### PR TITLE
Update api

### PR DESCRIPTION
**Pull request by domdinicola**

### CODE REVIEW

File: src/etools/applications/tpm/views.py [hunk 1]
```diff
@@ -167,10 +167,7 @@ def sync(self, request, *args, **kwargs):
         instance = queryset.filter(vendor_number=kwargs.get('vendor_number')).first()
 
         if not instance:
-            handler = TPMPartnerManualSynchronizer(
-                business_area_code=request.user.profile.country.business_area_code,
-                object_number=kwargs.get('vendor_number')
-            )
+            handler = TPMPartnerSynchronizer(vendor=kwargs.get('vendor_number'))
```
Comment (robertavram) [inline src/etools/applications/tpm/views.py:170]: are TPM partners global?

**Review by robertavram — COMMENTED** [review]
to discuss on a call tomorrow

File: src/etools/applications/funds/tasks.py [hunk 2]
```diff
@@ -10,8 +10,8 @@
 
 def sync_single_delegated_fr(business_area_code, fr_number):
     handler = DelegatedFundReservationsSynchronizer(
-        business_area_code=business_area_code,
-        object_number=fr_number
+        fr_number,
+        business_area_code=business_area_code
```
[the remark this answers is not in the document]
Comment (ntrncic) [reply src/etools/applications/funds/tasks.py:14]: it's still needed just tested without

File: src/etools/applications/partners/synchronizers.py [hunk 3]
```diff
@@ -1,12 +1,11 @@
-import json
 import logging
 from datetime import datetime
 from decimal import Decimal
 
 from django.core.exceptions import ValidationError
 from django.db import connection, transaction
 
-from unicef_vision.loaders import VISION_NO_DATA_MESSAGE
+from unicef_vision.loaders import INSIGHT_NO_DATA_MESSAGE
```
[the remark this answers is not in the document]
Comment (ntrncic) [reply src/etools/applications/partners/synchronizers.py:8]: removed.

File: src/etools/applications/partners/synchronizers.py [hunk 4]
```diff
@@ -88,7 +87,7 @@ class PartnerSynchronizer(VisionDataTenantSynchronizer):
     }
 
     def _convert_records(self, records):
-        return json.loads(records)['ROWSET']['ROW']
+        return records['ROWSET']['ROW']
```
[the remark this answers is not in the document]
Comment (ntrncic) [reply src/etools/applications/partners/synchronizers.py:90]: removed and still works

File: src/etools/applications/funds/synchronizers.py [hunk 5]
```diff
@@ -121,7 +121,7 @@ def _convert_records(self, records):
         # Since our counterparts are unable to return json for the json endpoints in case of 400+ or 500+ we should
         # catch known errors
         try:
-            json_records = json.loads(records)["ROWSET"]["ROW"]
+            json_records = records["ROWSET"]["ROW"]
```
[the remark this answers is not in the document]
Comment (ntrncic) [reply src/etools/applications/funds/synchronizers.py:124]: removed

File: src/etools/applications/funds/synchronizers.py [hunk 6]
```diff
@@ -387,13 +381,15 @@ def bad_record(record):
                 return False
             if not record['FC_NUMBER']:
                 return False
+            if not record['FR_NUMBER']:
```
[the remark this answers is not in the document]
Comment (ntrncic) [reply src/etools/applications/funds/synchronizers.py:384]: took it out

File: src/etools/applications/partners/synchronizers.py [hunk 7]
```diff
@@ -156,9 +155,10 @@ def _partner_save(self, partner, full_sync=True):
                 logger.info('Partner {} skipped, because PartnerType is {}'.format(
                     partner['VENDOR_NAME'], partner['PARTNER_TYPE_DESC']
                 ))
+
                 if partner_org.id:
-                    partner_org.deleted_flag = True if partner.get('MARKED_FOR_DELETION', None) else False
-                    partner_org.blocked = True if partner.get('POSTING_BLOCK', None) else False
+                    partner_org.deleted_flag = True if partner['MARKED_FOR_DELETION'] else False
```
[the remark this answers is not in the document]
Comment (ntrncic) [reply src/etools/applications/partners/synchronizers.py:160]: done

**Pull request merged**